src/main/java/net/minecraftforge/client/model/BlockStateLoader.java

```diff
@@ -98,10 +94,21 @@ public static BlockModelDefinition load(Reader reader, ResourceLocation location
                             boolean uvLock = var.getUvLock().orElse(false);
                             int weight = var.getWeight().orElse(1);
 
-                            if (var.isVanillaCompatible())
-                                mcVars.add(new Variant(var.getModel(), (ModelRotation)var.getState().orElse(ModelRotation.X0_Y0), uvLock, weight));
-                            else
-                                mcVars.add(new ForgeVariant(location, var.getModel(), var.getState().orElse(TRSRTransformation.identity()), uvLock, var.getSmooth(), var.getGui3d(), weight, var.getTextures(), var.getOnlyPartsVariant(), var.getCustomData()));
+                            ResourceLocation modelLocation = var.getModel();
+                            if (!var.isVanillaCompatible() && bakery != null)
+                            {
+                                modelLocation = new ResourceLocation(
+                                        "internal",
+                                        String.format("%d", internalGeneratedModelId++) + "/" + location.getNamespace() + "/" + location.getPath() + "/" +
```

Comment:
just do `String.valueOf` or even `"" +`

Reply:
Ah forgot String.valueOf. Yes that's better.